File: tests/fixtures/positive/java/CryptoUsage.java
package positive;

import java.security.KeyPairGenerator;
import java.security.MessageDigest;
import javax.crypto.Cipher;

/** Positive fixture for the JCA detectors. */
public class CryptoUsage {

    /** Expect AES-CBC with PKCS5 padding, from the transformation string. */
    public Cipher aesCbc() throws Exception {
        return Cipher.getInstance("AES/CBC/PKCS5Padding");
    }

    /** Expect AES-GCM. */
    public Cipher aesGcm() throws Exception {
        return Cipher.getInstance("AES/GCM/NoPadding");
    }

    /** Expect AES-ECB — a finding in its own right regardless of key size. */
    public Cipher aesEcb() throws Exception {
        return Cipher.getInstance("AES/ECB/PKCS5Padding");
    }

    /** Expect MD5. */
    public MessageDigest legacyDigest() throws Exception {
        return MessageDigest.getInstance("MD5");
    }

    /** Expect RSA key generation, with the size carried by initialize(). */
    public KeyPairGenerator rsaKeys() throws Exception {
        KeyPairGenerator generator = KeyPairGenerator.getInstance("RSA");
        generator.initialize(2048);
        return generator;
    }
}

File: tests/fixtures/positive/javascript/crypto_usage.js
// Positive fixture for the Node crypto detectors (not covered by CBOMkit).
const crypto = require("crypto");

// Expect MD5.
function legacyHash(data) {
  return crypto.createHash("md5").update(data).digest("hex");
}

// Expect SHA-256.
function modernHash(data) {
  return crypto.createHash("sha256").update(data).digest("hex");
}

// Expect AES-256-CBC, with the key size parsed out of the transformation.
function encrypt(key, iv, data) {
  const cipher = crypto.createCipheriv("aes-256-cbc", key, iv);
  return Buffer.concat([cipher.update(data), cipher.final()]);
}

// Expect RSA key generation.
function keys() {
  return crypto.generateKeyPairSync("rsa", { modulusLength: 2048 });
}

module.exports = { legacyHash, modernHash, encrypt, keys };
